### PR TITLE
Enhancement calculus improvements

### PR DESCRIPTION
**Pull request by 404Naz**
Fixed some existing problems. Added improvements for derivative and integral of logarithms. Adds a fix for #113.

### CODE REVIEW

**Review by matthew-mccall — CHANGES_REQUESTED** [review]

File: extras/src/MathMLSerializer.cpp [hunk 1]
```diff
@@ -67,27 +70,21 @@ void MathMLSerializer::Serialize(const Matrix& matrix)
     mrow->InsertEndChild(table);
     mrow->InsertEndChild(closeBrace);
 
-    //    // Integral symbol
-    //    tinyxml2::XMLElement* inte = doc.NewElement("mo");
-    //    inte->SetText("\u222B;");
```
Comment (matthew-mccall) [inline extras/src/MathMLSerializer.cpp:72]: Just remove the code if its not needed

File: extras/src/MathMLSerializer.cpp [hunk 2]
```diff
@@ -67,27 +70,21 @@ void MathMLSerializer::Serialize(const Matrix& matrix)
...
-    //    mrow->InsertEndChild(expElement);
-    //    mrow->InsertEndChild(dVar);
-
     result = mrow;
 }
 
+void MathMLSerializer::Serialize(const Oasis::Pi&)
+{
+    result = doc.NewElement("mi");
+    result->SetText("&pi");
```
Comment (matthew-mccall) [inline extras/src/MathMLSerializer.cpp:79]: It's `&pi;`

File: include/Oasis/Magnitude.hpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    explicit Magnitude(const OperandT& operand)
+        : UnaryExpression<Magnitude, OperandT>(operand)
+    {
+    }
+
+    [[nodiscard]] auto Simplify() const -> std::unique_ptr<Expression> override
+    {
+        auto simpOpU = this->GetOperand().Generalize();
```
Comment (matthew-mccall) [inline include/Oasis/Magnitude.hpp:42]: You don't have to generalize to turn call `Simplify`

**Review by matthew-mccall — CHANGES_REQUESTED** [review]

File: extras/src/MathMLSerializer.cpp (hunk 2 above)
Comment (404Naz) [reply extras/src/MathMLSerializer.cpp:79]: What is &pi?

File: extras/src/MathMLSerializer.cpp (hunk 1 above)
Comment (404Naz) [reply extras/src/MathMLSerializer.cpp:72]: It is removed... If it is there on the branch, I do not know how to remove it.

**Review by matthew-mccall — APPROVED** [review]
LGTM

**Pull request merged**